atomate/vasp/database.py

```diff
@@ -103,100 +111,57 @@ def insert_task(self, task_doc, use_gridfs=False):
 
-        # insert the task document
-        t_id = self.insert(task_doc)
+        big_data_to_store = {}
 
-        # insert the dos into gridfs and update the task document
-        if dos:
-            dos_gfs_id, compression_type = self.insert_gridfs(
-                dos, "dos_fs", task_id=t_id
-            )
-            self.collection.update_one(
-                {"task_id": t_id},
-                {"$set": {"calcs_reversed.0.dos_compression": compression_type}},
-            )
-            self.collection.update_one(
-                {"task_id": t_id}, {"$set": {"calcs_reversed.0.dos_fs_id": dos_gfs_id}}
-            )
+        def extract_from_calcs_reversed(obj_key):
+            """
+            Grab the data from calcs_reversed.0.obj_key and store on gridfs directly or some Maggma store
+            Args:
+                obj_key: Key of the data in calcs_reversed.0 to store
+            """
+            if obj_key in task_doc["calcs_reversed"][0]:
+                calcs_r_data = task_doc["calcs_reversed"][0][obj_key]
+            else:
+                calcs_r_data = task_doc["calcs_reversed"][0][obj_key]
 
-        # insert the bandstructure into gridfs and update the task documents
-        if bs:
-            bfs_gfs_id, compression_type = self.insert_gridfs(
-                bs, "bandstructure_fs", task_id=t_id
-            )
-            self.collection.update_one(
-                {"task_id": t_id},
-                {
-                    "$set": {
-                        "calcs_reversed.0.bandstructure_compression": compression_type
-                    }
-                },
-            )
-            self.collection.update_one(
-                {"task_id": t_id},
-                {"$set": {"calcs_reversed.0.bandstructure_fs_id": bfs_gfs_id}},
-            )
+            big_data_to_store[obj_key] = calcs_r_data
+
+            # remove the big object from all calcs_reversed
+            # this can catch situations were the drone added the data to more than one calc.
+            for i_calcs in range(len(task_doc["calcs_reversed"])):
+                del task_doc["calcs_reversed"][i_calcs][obj_key]
+
+        # drop the data from the task_document and keep them in a separate dictionary (big_data_to_store)
+        if self._maggma_store_type in VALID_STORES or use_gridfs and "calcs_reversed" in task_doc:
+            for data_key in OBJ_NAMES:
+                if data_key in task_doc["calcs_reversed"][0]:
+                    extract_from_calcs_reversed(data_key)
 
-        # insert the CHGCAR file into gridfs and update the task documents
-        if vol_data:
-            for name, data in vol_data.items():
-                data_gfs_id, compression_type = self.insert_gridfs(
-                    data, "{}_fs".format(name), task_id=t_id
-                )
+        # insert the task document
+        t_id = self.insert(task_doc)
+
+        if "calcs_reversed" in task_doc:
+            #
```

Comment:
I think this whole if/else block should just be handled by a `self.insert_object` function or something similar.

That function would then just intelligently select which insertion method to use based on the value of self._maggma_store_type.

Also it seems to me that any maggma store should be supported for inserts right? so the NotImplementedError is not needed perhaps?